scratchpad/tn_api/tn.py

```diff
@@ -52,16 +53,21 @@ class Port:
 class TensorNetwork(TensorNetworkIFC[np.ndarray]):
     tensors: Iterable[np.ndarray]
     shape: tuple
-    edges: tuple[tuple[Port]]
+    edges: tuple
 
     def __init__(self, *args, **kwargs):
         self._tensors = []
         self._edges = tuple()
         self.shape = tuple()
+        self.buckets = []
+        self.data_dict = {}
 
     # slice not inplace
     def slice(self, slice_dict: dict) -> 'TensorNetwork':
-        ...
+        tn = self.copy()
+        sliced_buckets = np_framework.get_sliced_np_buckets(self.buckets, self.data_dict, slice_dict)
```

Comment:
i imagine `data_dict` needs to be populated for this to be meaningful logic. is that something the callers should pass in? @danlkv